test/required/requiredexample.proto

```diff
@@ -0,0 +1,14 @@
+package required;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+message RequiredExample {
+    required string theRequiredString = 1;
+    optional string theOptionalString = 2;
+    repeated string theRepeatedStrings = 3;
+}
```

Comment:
Please copy test.NidOptNative and test.NinOptNative and change optional to required.

Create a test that imports test.NidOptNative and calls test.NewPopulatedNidOptNative and Marshal the message.
Then take those bytes and Unmarshal them into required.NidOptNative.
This should never return an error.

Create another test that imports test.NinOptNative and calls test.NewPopulatedNinOptNative.
Now this might be a challenge.
I don't know how familiar you are with reflect, but I will write this test myself if you say that you don't have enough time.
Using reflect set a random field to Nil.
Then Marshal test.NinOptNative and Unmarshal the bytes into required.NinOptNative.
This should always return an error.

Finally Create a test using required.NinOptNative and required.NewPopulatedNinOptNative.
Again set a random field to Nil, easiest with reflect.
Then Marshal, this should always return an error.